crypto/bio/bss_conn.c

```diff
@@ -595,6 +596,56 @@ static int conn_puts(BIO *bp, const char *str)
     return ret;
 }
 
+int conn_gets(BIO *bio, char *buf, int size)
+{
+    BIO_CONNECT *data;
+    char *ptr = buf;
+    int ret = 0;
+
+    if (buf == NULL) {
+        ERR_raise(ERR_LIB_BIO, ERR_R_PASSED_NULL_PARAMETER);
+        return -1;
+    }
+    if (size <= 0) {
+        ERR_raise(ERR_LIB_BIO, BIO_R_INVALID_ARGUMENT);
+        return -1;
+    }
+    *buf = '\0';
+
+    if (bio == NULL || bio->ptr == NULL) {
```

Comment:
merge this into 605?

Reply:
I deliberately left them apart (as done also for `BIO_get_line()`)
because as soon as input validation of `buf` and `size` is successful, ` *buf = '\0';` can be done,
which is defensive in case an error is returned due to `bio == NULL || bio->ptr == NULL` but ignored by the caller.